core/services/ocr2/plugins/functions/config/config_types.proto

```diff
@@ -9,11 +9,26 @@ enum AggregationMethod {
     AGGREGATION_MEDIAN = 1;
 }
 
+message OracleIDtoKeyShareIndex {
+  uint32 oracle_id = 1;
+  uint32 key_share_index = 2;
+}
+
+message ThresholdReportingPluginConfig {
+  uint32 max_query_length_bytes = 1;
+  uint32 max_observation_length_bytes = 2;
+  uint32 max_report_length_bytes = 3;
+  uint32 request_count_limit = 4;
+  uint32 request_total_bytes_limit = 5;
+  bool require_local_request_check = 6;
+}
+
 message ReportingPluginConfig {
     uint32 maxQueryLengthBytes = 1;
     uint32 maxObservationLengthBytes = 2;
     uint32 maxReportLengthBytes = 3;
     uint32 maxRequestBatchSize = 4;
     AggregationMethod defaultAggregationMethod = 5;
     bool uniqueReports = 6;
+    ThresholdReportingPluginConfig thresholdPluginConfig = 7;
```

Comment:
Let's make this an optional field.

Reply:
According to the proto3 standard, all fields are optional by default, unlike proto2 where it had to be explicitly specified that a field is optional.

Reply:
oh cool!